Common/File/Path.cpp

```diff
@@ -117,6 +118,16 @@ std::string Path::GetFileExtension() const {
 
 std::string Path::GetDirectory() const {
 	size_t pos = path_.rfind('/');
+	if (type_ == PathType::HTTP) {
+		// Things are a bit different for HTTP, because we probably ended with /.
+		if (pos + 1 == path_.size()) {
+			pos = path_.rfind('/', pos - 1);
+			if (pos != path_.npos && pos > 8) {
+				return path_.substr(0, pos + 1);
+			}
+		}
+	}
+
```

Comment:
Starting to think that stripping the / for regular directories might be a mistake and we'd get simpler code if we just always have it on directory paths. Though at least now we have a centralized place to fix that :)